hal/common/critical.c

```diff
@@ -233,10 +233,10 @@ bool core_util_atomic_cas_u32(uint32_t *ptr, uint32_t *expectedCurrentValue, uin
 }
 
 bool core_util_atomic_cas_ptr(void **ptr, void **expectedCurrentValue, void *desiredValue) {
```

Comment:
Ah! Good catch! This explains why none of the warnings/errors were being reported before.

Should be fixed now.